components/blitz/src/omero/gateway/facility/TablesFacilityHelper.java

```diff
@@ -0,0 +1,457 @@
+            }
+            if (col instanceof ImageColumn) {
+                ImageData[] rowData = new ImageData[nRows];
+                long tableData[] = ((ImageColumn) col).values;
+                for (int j = 0; j < nRows; j++) {
+                    Image im = new ImageI(tableData[j], false);
+                    rowData[j] = new ImageData(im);
+                }
+                dataArray[i] = rowData;
+                header[i].setType(ImageData.class);
+            }
+            if (col instanceof LongArrayColumn) {
+                Long[][] rowData = new Long[nRows][];
+                long tableData[][] = ((LongArrayColumn) col).values;
+                for (int j = 0; j < nRows; j++) {
+                    Long[] tmp = new Long[tableData[j].length];
+                    for (int k = 0; k < tableData[j].length; k++) {
+                        tmp[k] = tableData[j][k];
+                    }
+                    rowData[j] = tmp;
+                }
+                dataArray[i] = rowData;
+                header[i].setType(Long[].class);
+            }
+            if (col instanceof LongColumn) {
+                Long[] rowData = new Long[nRows];
+                long tableData[] = ((LongColumn) col).values;
+                for (int j = 0; j < nRows; j++)
+                    rowData[j] = tableData[j];
+                dataArray[i] = rowData;
+                header[i].setType(Long.class);
+            }
+            if (col instanceof MaskColumn) {
+                MaskColumn mc = ((MaskColumn) col);
+                MaskData[] rowData = new MaskData[nRows];
+                for (int j = 0; j < nRows; j++) {
+                    MaskData md = new MaskData(mc.x[j], mc.y[j], mc.w[j],
+                            mc.h[j], mc.bytes[j]);
+                    rowData[j] = md;
+                }
+                dataArray[i] = rowData;
+                header[i].setType(MaskData.class);
+            }
+            if (col instanceof PlateColumn) {
+                PlateData[] rowData = new PlateData[nRows];
+                long tableData[] = ((PlateColumn) col).values;
+                for (int j = 0; j < nRows; j++) {
+                    Plate p = new PlateI(tableData[j], false);
+                    rowData[j] = new PlateData(p);
+                }
+                dataArray[i] = rowData;
+                header[i].setType(PlateData.class);
+            }
+            if (col instanceof RoiColumn) {
+                ROIData[] rowData = new ROIData[nRows];
+                long tableData[] = ((RoiColumn) col).values;
+                for (int j = 0; j < nRows; j++) {
+                    Roi p = new RoiI(tableData[j], false);
+                    rowData[j] = new ROIData(p);
+                }
+                dataArray[i] = rowData;
+                header[i].setType(ROIData.class);
+            }
+            if (col instanceof StringColumn) {
+                String[] rowData = ((StringColumn) col).values;
```

Comment:
good be simplified but very minor i.e. ``dataArray[i] = ((StringColumn) col).values;``